RELEASES.md

```diff
@@ -197,11 +199,107 @@ To create a minor release:
+harness tests are run repeatedly and a variety of in-house testnets are run to
+ensure Tendermint functions at the scale it will be used by application
+developers and node operators.
+
+### Nightly End-To-End Tests
+
+The Tendermint team maintains [a set of end-to-end
+tests](https://github.com/tendermint/tendermint/blob/master/test/e2e/README.md#L1)
+that run each night on the latest commit of the project and on the code in the
+tip of each backport branch. These tests start a network of containerized
+Tendermint processes and run automated checks that the network functions as
+expected in both stable and unstable conditions. During the feature freeze,
+these tests are run nightly and must pass consistently for a release of
+Tendermint to be considered stable.
+
+### Upgrade Harness
+
+The Tendermint team is creating an upgrade test harness to exercise the
+workflow of stopping an instance of Tendermint running one version of the
+software and starting up the same application running the next version. To
+support upgrade testing, we will add the ability to terminate the Tendermint
+process at specific pre-defined points in its execution so that we can verify
+upgrades work in a representative sample of stop conditions.
+
+### Large Scale Testnets
+
+The Tendermint end-to-end tests run a small network (~10s of nodes) to exercise
+basic consensus interactions. Real world deployments of Tendermint often have over 
+a hundred nodes just in the validator set, with many others acting as full
+nodes and sentry nodes. To gain more assurance before a release, we will also run
+larger-scale test networks to shake out emergent behaviors at scale.
+
+Large-scale test networks are run on a set of virtual machines (VMs). Each VM
+is equipped with 4 Gigabytes of RAM and 2 CPU cores. The network runs a very
+simple key-value store application. The application adds artificial delays to
+different ABCI calls to simulate a slow application. During each test net, the
+following metrics are monitored and collected on each node:
+
+* Tendermint rounds per height
+* Peers connected
+* Memory resident set size
+* CPU utilization
+* Blocks produced per minute
+* Seconds for each step of consensus (Propose, Prevote, Precommit, Commit)
+* Latency to receive each block proposal
+
+For these tests we intentionally target low-powered host machines (with low core
+counts and limited memory) to ensure we observe similar kinds of resource contention 
+and limitation that real-world  deployments of Tendermint experience in production. 
+
+#### 200 Node Testnet
+
+To test the stability and performance of Tendermint in a real world scenario,
+a 200 node test network is run. The network comprises 5 seed nodes, 100
+validators and 95 non-validating full nodes. All nodes begin by dialing
+a subset of the seed nodes to discover peers. The network is run for several
+days, with metrics being collected continuously.
```

Comment:
What are the targets for each of the metrics outlined above? I think we need to take a first stab at defining the values of a handful of those metrics to be able to know, with evidence, that the release is stable.

Reply:
Yes, my point is that I want to see _numbers_ associated with each of those metrics :slightly_smiling_face: i.e. targets

Reply:
It's valuable to distinguish between _indicators_ and _objectives_. Which metrics we are going to track (an example of indicators) is probably worth including in a general doc like this. Specific target values (i.e., objectives) are harder to estimate, and I don't think we should hold off on making any plans for the detailed task of figuring out what values to aim for. We can always add them later (and should), and in the meanwhile I think there's some communicative utility in documenting the outline of the plan.

Reply:
Where do we then list the detailed objectives against which we're going to be testing releases? Ideally with some rational justification for the objectives. We need to write this down/communicate/agree on this somewhere.

Ultimately the artifact I'd like to see here would be a report that we can attach to each release to say, "this is the set of tests and objectives against which we tested this release in order to deem it release-worthy". The first draft test strategy will never be perfect, and we can naturally evolve the specific objectives and indicators over time as we gain insight into users' workloads and update our testing strategy accordingly.